parser_library/fuzzer/fuzzer.cpp

```diff
@@ -96,7 +118,17 @@ extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size)
     }
     *target = hlasm_plugin::utils::replace_non_utf8_chars(std::string_view((const char*)data, size));
 
-    analyzer a(source, analyzer_options(&lib, db2_preprocessor_options()));
+    return source;
+}
+} // namespace
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size)
+{
+    if (!data || size == 0)
+        return 0;
+
+    fuzzer_lib_provider lib;
+    analyzer a(get_content(data, size, lib), analyzer_options(&lib, get_preprocessor_options(std::bitset<3>(data[0]))));
```

Comment:
the return value from get_content needs to be captured in a variable...

And I still think that the first byte should be skipped... the preprocessors are sensitive to various directives that typically start with `*` and that means only one particular combination would be tested.
```suggestion
    auto source = get_content(data + 1, size - 1, lib);
    analyzer a(source, analyzer_options(&lib, get_preprocessor_options(std::bitset<3>(*data))));
```